app-editors/zed/zed-0.150.4.ebuild

```diff
@@ -0,0 +1,1161 @@
+	[xim]='https://github.com/npmania/xim-rs;27132caffc5b9bc9c432ca4afad184ab6e7c16af;xim-rs-%commit%'
+	[xkbcommon]='https://github.com/ConradIrwin/xkbcommon-rs;fcbb4612185cc129ceeff51d22f7fb51810a03b2;xkbcommon-rs-%commit%'
+)
+
+LLVM_COMPAT=( 18 )
+
+inherit cargo check-reqs desktop flag-o-matic llvm-r1 toolchain-funcs xdg
+
+DESCRIPTION="The fast, collaborative code editor"
+HOMEPAGE="https://zed.dev https://github.com/zed-industries/zed"
+SRC_URI="
+	https://github.com/zed-industries/zed/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	${CARGO_CRATE_URIS}"
+
+LICENSE="0BSD AGPL-3+ Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD BSD-2 Boost-1.0 CC0-1.0 GPL-3+ ISC LGPL-3 MIT MPL-2.0 Unicode-DFS-2016 Unlicense ZLIB"
+SLOT="0"
+KEYWORDS="~amd64"
+CHECKREQS_DISK_BUILD="8G"
+
+DEPEND="
+	app-arch/zstd:=
+	dev-db/sqlite:3
+	dev-libs/libgit2:=
+	dev-libs/mimalloc
+	dev-libs/openssl:0/3
+	dev-libs/wayland
+	dev-libs/wayland-protocols
+	dev-util/wayland-scanner
+	dev-util/vulkan-tools
+	media-libs/alsa-lib
+	media-libs/fontconfig
+	media-libs/vulkan-loader[X]
+	net-misc/curl
+	sys-libs/zlib
+	x11-libs/libxcb:=
+	x11-libs/libxkbcommon[X]
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	>=virtual/rust-1.80.0
+	dev-util/vulkan-headers
+	sys-devel/gettext
+	sys-devel/mold
+	$(llvm_gen_dep '
+		sys-devel/clang:${LLVM_SLOT}=
+		sys-devel/llvm:${LLVM_SLOT}=
+	')
+"
+
+QA_FLAGS_IGNORED="usr/bin/zed"
+
+pkg_setup() {
+	if tc-is-gcc; then
+		export CARGO_PROFILE_RELEASE_LTO="true"
+	elif tc-is-clang; then
+		export CARGO_PROFILE_RELEASE_LTO="thin"
+	fi
+	strip-unsupported-flags
+	# flags from upstream
+	export RUSTFLAGS="${RUSTFLAGS} -C symbol-mangling-version=v0 --cfg tokio_unstable -C link-arg=-fuse-ld=mold -C link-args=-Wl,--disable-new-dtags,-rpath,\$ORIGIN/../lib"
+	# linking error with llvm-18
+	export RUSTFLAGS="${RUSTFLAGS} -C link-args=-Wl,-z,nostart-stop-gc"
+}
+
+src_prepare() {
+	default
+
+	# Cargo offline fetch workaround
+	local TS_COMMIT="7f4a57817d58a2f134fe863674acad6bbf007228"
+	local TS_GIT="git = \"https://github.com/tree-sitter/tree-sitter\", rev = \"${TS_COMMIT}\""
+	local TS_PATH="path = \"${WORKDIR}/tree-sitter-${TS_COMMIT}/lib\""
+
+	sed -e "s#${TS_GIT}#${TS_PATH}#" \
+		-i "${S}/Cargo.toml" || die "Cargo fetch workaround failed"
+}
+
+src_configure() {
+	cargo_src_configure --all-features
+}
+
+src_compile() {
+	export RELEASE_VERSION="${PV}"
+	export ZED_UPDATE_EXPLANATION='Updates are handled by portage'
+	cargo_src_compile --package zed --package cli
+}
+
+src_install() {
+	newbin $(cargo_target_dir)/cli zed
+	exeinto "/usr/libexec"
+	newexe $(cargo_target_dir)/zed zed-editor
+
+	newicon -s 512 crates/zed/resources/app-icon.png zed.png
+	newicon -s 1024 crates/zed/resources/app-icon@2x.png zed.png
+	make_desktop_entry /usr/bin/zed Zed zed.png "TextEditor;Development;IDE"
+}
+
+src_test () {
+	mkdir -p ${HOME}/.config/zed
+	mkdir -p ${HOME}/.local/share/zed/logs/
```

Comment:
These should use `|| die` then, for standard stylistic reasons.

Yes, I know, the consequences of failing to create a directory that the tests rely on to pass would simply mean the tests fail to pass, which is an error either way. But... them's the rules.

Reply:
At least it stops immediately and doesn't waste time even trying!